Extensions/Video/videoruntimeobject.js

```diff
@@ -0,0 +1,241 @@
+  this._property1 = objectData.content.property1;
+
+  // Create the renderer (see videoruntimeobject-pixi-renderer.js)
+  if (this._renderer)
+    gdjs.VideoRuntimeObjectRenderer.call(this._renderer, this, runtimeScene);
+  else this._renderer = new gdjs.VideoRuntimeObjectRenderer(this, runtimeScene);
+};
+
+gdjs.VideoRuntimeObject.prototype = Object.create(gdjs.RuntimeObject.prototype);
+gdjs.VideoRuntimeObject.thisIsARuntimeObjectConstructor = "Video::VideoObject"; //Replace by your extension + object name.
+
+gdjs.VideoRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+/**
+ * Called once during the game loop, before events and rendering.
+ * @param {gdjs.RuntimeScene} runtimeScene The gdjs.RuntimeScene the object belongs to.
+ */
+gdjs.VideoRuntimeObject.prototype.update = function(runtimeScene) {
+  // This is an example: typically you want to make sure the renderer
+  // is up to date with the object.
+  this._renderer.ensureUpToDate();
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.VideoRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    this.setWidth(initialInstanceData.width);
+    this.setHeight(initialInstanceData.height);
+  }
+};
+
+/**
+ * Update the object position.
+ * @private
+ */
+gdjs.VideoRuntimeObject.prototype._updatePosition = function() {
+  // This is an example: typically you want to tell the renderer to update
+  // the position of the object.
+  this._renderer.updatePosition();
+};
+
+/**
+ * Set object position on X axis.
+ */
+gdjs.VideoRuntimeObject.prototype.setX = function(x) {
+  gdjs.RuntimeObject.prototype.setX.call(this, x); // Always call the parent method first.
+  this._updatePosition();
+};
+
+/**
+ * Set object position on Y axis.
+ */
+gdjs.VideoRuntimeObject.prototype.setY = function(y) {
+  gdjs.RuntimeObject.prototype.setY.call(this, y); // Always call the parent method first.
+  this._updatePosition();
+};
+
+/**
+ * Set the angle of the object.
+ * @param {number} angle The new angle of the object
+ */
+gdjs.VideoRuntimeObject.prototype.setAngle = function(angle) {
+  gdjs.RuntimeObject.prototype.setAngle.call(this, angle); // Always call the parent method first.
+  this._renderer.updateAngle(); // Tell the renderer to update the rendered object
+};
+
+/**
+ * Set object opacity.
+ */
+gdjs.VideoRuntimeObject.prototype.setOpacity = function(opacity) {
+  if (opacity < 0) opacity = 0;
+  if (opacity > 255) opacity = 255;
+  var new_opacity = opacity / 255;
+  //this.opacity = opacity || this.getOpacity() ;
+  this._renderer.updateOpacity(new_opacity); // Tell the renderer to update the rendered object
+};
+
+/**
+ * Get object opacity.
+ * return 0-1
+ */
+gdjs.VideoRuntimeObject.prototype.getOpacity = function() {
+  return this._renderer.getOpacity() * 255;
```

Comment:
Prefer keeping all the state in the object:

`return this.opacity`

renderer should only keep the bare minimum that is needed for rendering (most of the time, just a pixi object).